velox/serializers/PrestoSerializer.cpp

```diff
@@ -995,6 +998,7 @@ class VectorStream {
   int32_t nullCount_{0};
   int32_t totalLength_{0};
   bool hasLengths_{false};
+  bool counted_{false};
```

Comment:
The name of this member variable is not self-descriptive. It is not clear what is being counted here. 

I wonder if a better fix would be to write 0 into length_ at construction time for  ROW, ARRAY and MAP and remove that logic from appendLength. This would make flush method non-mutating.

```
  void appendLength(int32_t length) {
    if (nullCount_ + nonNullCount_ == 1) {
      // The first element in the offsets in the wire format is always 0 for
      // nested types but not for string.
      auto kind = type_->kind();
      if (kind == TypeKind::ROW || kind == TypeKind::ARRAY ||
          kind == TypeKind::MAP) {
        lengths_.appendOne<int32_t>(0);
      }
    }
    totalLength_ += length;
    lengths_.appendOne<int32_t>(totalLength_);
  }
```